mdsobjects/cpp/mdstreeobjects.cpp

```diff
@@ -556,6 +556,26 @@ EXPORT void TreeNode::operator delete(void *p)
     ::operator delete(p);
 }
 
+std::string  TreeNode::getNciString(int itm)
```

Comment:
I must do this way because the derived TreeNodeThinClient redefines getnci so that it can reuse a lot of TreeNode code.  Unfortunately it is not possible having a virtual method that is also a template. So the only way was to make it a bit  less elegant in TreeNode and providing three methods for getNci.